src/main/java/seedu/address/model/shift/RoleRequirement.java

```diff
@@ -46,6 +46,9 @@ public RoleRequirement(String roleRequirementInfo) {
      * Returns true if a given string is a valid role requirement
      */
     public static boolean isValidRoleRequirement(String test) {
+        if (test.trim() == "") {
```

Comment:
```suggestion
        if (test.trim().isEmpty()) {
```

Reply:
I removed this check in favor of just using regex only, the empty string is already rejected by the regex as seen in the RoleRequirementTest

Reply:
@WangZijun97 Yup, I have added the test cases and the regex does reject whitespace cases